Python/pytime.c

```diff
@@ -301,9 +315,16 @@ _PyTime_FromTimeval(_PyTime_t *tp, struct timeval *tv, int raise)
         }
         res = -1;
     }
-    t = t * SEC_TO_NS;
+    t *= SEC_TO_NS;
 
-    t += (_PyTime_t)tv->tv_usec * US_TO_NS;
+    if (_PyTime_FromMicroseconds(&us, tv->tv_usec) < 0) {
+        if (!raise) {
```

Comment:
Isn't it too late? You've already raised the exception.

Reply:
I removed _PyTime_FromMicroseconds(), it was a bad idea.